### ISSUE TITLE
Clarification about installing and using this plugin

### ISSUE THREAD

**Issue by ajtruckle** [EnlighterJS/Plugin.TinyMCE#2]
I have several plugins installed with respects to TinyMCE.

- TinyMCE Advanced

I use the above plugin for core formatting of the TinyMCE toolbar, adding buttons that I would like displayed and some default editor behaviour.

- Advanced TinyMCE Configuration

I use the above plugin for additional configuration. Specifically, for:

```
content_style
.mce-content-body{font:12pt Tahoma, sans-serif}
```

- Add AutoSave | Fullscreen Plugins to TinyMCE

The above is my own plugin and will install the TimyMCE AutoSave | Fullscreen Plugins (version 4.9.6) and register them with TinyMCE. It is a simple plugin:

```
<?php
/**
* Plugin Name: Add AutoSave | Fullscreen Plugins to TinyMCE
* Plugin URI: https://www.trucklesoft.co.uk/
* Description: This will install the TimyMCE AutoSave | Fullscreen Plugins (version 4.9.6) and register them with TinyMCE.
* Version: 1.2
* Author: Andrew Truckle
* Author URI: https://www.trucklesoft.co.uk/
**/

/**
 * Add the TinyMCE AutoSave Plugin.
 *
 * @param array $plugins An array of all plugins.
 * @return array
 */
function tsoft_custom_plugins( $plugins ) {
     $plugins['autosave'] = plugins_url( 'mce/', __FILE__ ) . 'autosave/plugin.min.js';
     $plugins['fullscreen'] = plugins_url( 'mce/', __FILE__ ) . 'fullscreen/plugin.min.js';
     return $plugins;
}
add_filter( 'mce_external_plugins', 'tsoft_custom_plugins' );
```

So, we come you your **TinyMCE** plugin for **EnlighterJS**:

1. Do you foresee any compatibility issues with installing and activating this plugin?
2. Can you confirm if it is compatible with **bbPress** environment?

**Comment by AndiDittrich** [comment]
i know that's not the answer you would like to have...but in the world of open-source and especially WordPress it is nearly impossible to give you a valid or reliable answer to your question...

try+error is the recommend method to get mostly valid results for the specific combination of the active software versions - the results may change even between minor releases...

**Comment by ajtruckle** [comment]
Actually, according to your documentation for the EnlighterJS WordPress Plug-in the TinyMCE support is already there. I guess I have to manually activate it on my toolbar.

**Comment by AndiDittrich** [comment]
maybe one of your TinyMCE plugins is removing it from the toolbar - just disable all other plugins and find out which plugin causes the issue

**Comment by ajtruckle** [comment]
I tried disabling all of those plugins (including bbp style pack) and whilst it reset the toolbar to a basic version, your button was not there.

What I would like to try and do is manually load your button etc by extending my plugin with suitable code. I was trying to find what to include by examining your code but I even find the TinyMCE "code" plugin data.

**Comment by AndiDittrich** [comment]
don't try to do it manually, it's a lot more difficult then "just include" the js/css files.
normally it works out of the box and i've never experienced any issues with it

is the support enabled within the Enlighter settings page (Enlighter -> VIsual Editor -> TinyMCE) ?

**Comment by ajtruckle** [comment]
I notice that **Enable Frontend Integration** is unticked. Is this why? Since technically using the support forum is going to be in the front end.

**Comment by AndiDittrich** [comment]
because TinyMCE is not enabled on bbPress nor any other frontend plugin by default

**Comment by ajtruckle** [comment]
I tried switching it on (with all the plugins deactivated) and it insists in a basic TinyMCE toolbar. This is why I had to use the other plugins to get the extra toolbar features.

> don't try to do it manually, it's a lot more difficult then "just include" the js/css files.
> normally it works out of the box and i've never experienced any issues with it

I appreciate that and I don't particularly want to do anything manually. I just wish I could use it. Up until now I am remembering to put my editor into the "Text" mode before creating or editing a topic / reply that has code it in and that works fine (with exception of your toolbar etc not showing - but you have already explained why that is happening).

So when I saw the you also had a feature for using EnlghterJS within TinyMCE I thought it would be great to use that too. I remember too that you said not to toggle between one editor and the other when you had code in it and t stick to one or the other.

But right now, I need a way to load your button / system into my toolbar. All of what I want is in there (in bbPress) by using the plugins I already stated:

![Toolbar](https://user-images.githubusercontent.com/7985331/74727129-97ffee80-5238-11ea-8ffe-37cb0fe4a022.png)

**Comment by AndiDittrich** [comment]
is this a screenshot of your frontend or backend ? do you have the issues in your frontend or backend ? 

especially on the frontend you have to enable the TinyMCE editor for comments/bbpress manually by adding some code - in case this is not done the "right way" WordPress won't load any of the registered plugins and they didn't become visible...

**Comment by ajtruckle** [comment]
> is this a screenshot of your frontend or backend ?

This is a screenshot of the **frontend**. Any user of my forum will have access to create topics and add replies using either the visual or text editors.

> do you have the issues in your frontend or backend ?

I can only comment on the frontend because I don't actually know where in the backend the TinyMCE toolbar is used anymore. When I create pages and posts it uses Gutenberg. Can you suggest how I can test the **backend** for you?
 
> especially on the frontend you have to enable the TinyMCE editor for comments/bbpress manually by adding some code - in case this is not done the "right way" WordPress won't load any of the registered plugins and they didn't become visible...

As I mention, TinyMCE is already enabled for bbPress in the front end for several months now.

**Comment by AndiDittrich** [comment]
i understand that you've "enabled" TinyMCE for bbPress - but this doesn't mean that your code takes care of additional plugins - i assume that's the issue.

**Comment by ajtruckle** [comment]
Correct, that is why I installed **TinyMCE Advanced** because it installs any needed plugins and activates them for the toolbar.

And then, for the other two I created my own plugin which installs those TinyMCE plugins and activates them.

This is why I suggested that I we most likely need to manually activate your TinyMCE plugin similar to how I have done it so that it would work with my toolbar.

**Comment by AndiDittrich** [comment]
see https://github.com/EnlighterJS/documentation/blob/master/wordpress/TinyMCE.md#manual-integration

**Comment by AndiDittrich** [comment]
btw. did you use an admin account or an regular user account ? background: the frontend editing capabilities are restricted to admin+manager users per default for security reasons...

see https://github.com/EnlighterJS/documentation/blob/master/wordpress/FilterHooks.md#filterenlighter_frontend_editing

**Comment by ajtruckle** [comment]
> btw. did you use an admin account or an regular user account ?

At the time I was testing with my admin account. But standard account shows the same toolbar.



> see https://github.com/EnlighterJS/documentation/blob/master/wordpress/TinyMCE.md#manual-integration

Thanks for the link. I am going to have to leave this because at no time do I personally execute `tinymce.init` or `wp.editor.initialize`. All I do is hook into a filter:

```
function tsoft_custom_plugins( $plugins ) {
     $plugins['autosave'] = plugins_url( 'mce/', __FILE__ ) . 'autosave/plugin.min.js';
     $plugins['fullscreen'] = plugins_url( 'mce/', __FILE__ ) . 'fullscreen/plugin.min.js';
     return $plugins;
}
add_filter( 'mce_external_plugins', 'tsoft_custom_plugins' );
```

So I think I will have to just leave this and use the **text** editor.

**Comment by AndiDittrich** [comment]
i've investigated the issue - bbPress uses a minimal version of TinyMCE which removes all external plugins (it is not possible to change that behaviour). instead you can force bbPress to use the full TinyMCE version including all buttons:

```php
add_filter('bbp_after_get_the_content_parse_args', function($args = array()){
      $args['tinymce'] = true;
      $args['teeny'] = false;

      $args['quicktags'] = false;
      return $args;
});
```

i remember that there was a similar issue in the past but i didn't found the ist

## Example ##

![image](https://user-images.githubusercontent.com/1474107/74731574-86264780-5248-11ea-8b1c-3f1fda832dcf.png)
